cmake/modules/SwiftSource.cmake

```diff
@@ -451,12 +453,22 @@ function(_compile_swift_files
 
   # Then we can compile both the object files and the swiftmodule files
   # in parallel in this target for the object file, and ...
+
+  # Windows doesn't support long command line paths, of 8191 chars or over.
+  # We need to work around this by avoiding long command line arguments. This can be
+  # achieved by writing the list of file paths to a file, then reading that list
+  # in the Python script.
+  string(RANDOM file_name)
+  set(file_path "${CMAKE_CURRENT_BINARY_DIR}/${file_name}.txt")
+  list_join("${source_files}" ";" source_files_string)
```

Comment:
`;` is legal on Unix too. I'm pretty sure cmake is hopelessly doomed if any file path actually uses `;` so we don't need to work around that threat here.